kafka_consumer/datadog_checks/kafka_consumer/client/confluent_kafka_client.py

```diff
@@ -168,24 +167,77 @@ def _get_topic_partitions(self, topics, consumer_group):
 
             partitions = list(topics.topics[topic].partitions.keys())
 
-            for partition in partitions:
-                # Get all topic-partition combinations allowed based on config
-                # if topics is None => collect all topics and partitions for the consumer group
-                # if partitions is None => collect all partitions from the consumer group's topic
-                if not self.config._monitor_unlisted_consumer_groups and self.config._consumer_groups.get(
-                    consumer_group
-                ):
-                    if (
-                        self.config._consumer_groups[consumer_group]
-                        and topic not in self.config._consumer_groups[consumer_group]
+            if self.config._monitor_unlisted_consumer_groups:
+                for partition in partitions:
+                    self.log.debug("TOPIC PARTITION: %s", TopicPartition(topic, partition))
+                    yield TopicPartition(topic, partition)
+
+            elif self.config._consumer_groups_regex:
+                filtered_topic_partitions = self._get_filtered_topic_partitions(consumer_group, topic, partitions)
+                for filtered_topic_partition in filtered_topic_partitions:
+                    self.log.debug(
+                        "TOPIC PARTITION: %s", TopicPartition(filtered_topic_partition[0], filtered_topic_partition[1])
+                    )
+                    yield TopicPartition(filtered_topic_partition[0], filtered_topic_partition[1])
+
+            if self.config._consumer_groups:
```

Comment:
Yup, in the case where the same topic partition would be returned for both `consumer_groups` and `consumer_groups_regex`, the metrics wouldn't be affected since we wouldn't return duplicate metrics.

Reply:
I've also just added a new test to verify this.

Reply:
I could use a set to only return unique topic partition combinations, although I think the likelihood of a customer using both `consumer_groups` and `consumer_groups_regex` with overlapping topic partitions is low that maintaining this set might not be worthwhile?

Reply:
sgtm